src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -185,26 +192,58 @@ private void drawLineWithTranslate(Graphics2D graphics, Line2D line, double xOff
    * 
    * @param routeDescription {@linkplain RouteDescription} containing the Route information
    * @param mapData {@linkplain MapData} Object containing Information about the Map Coordinates
+   * @param width The width of the Map
+   * @param height The height of the Map
    * @return The {@linkplain Point} array specified by the {@linkplain RouteDescription} and {@linkplain MapData}
    *         objects
    */
-  protected Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData) {
+  protected Point[] getRoutePoints(RouteDescription routeDescription, MapData mapData, int xOffset, int yOffset,
+      int width, int height) {
     final List<Territory> territories = routeDescription.getRoute().getAllTerritories();
     final int numTerritories = territories.size();
     final Point[] points = new Point[numTerritories];
     for (int i = 0; i < numTerritories; i++) {
-      points[i] = mapData.getCenter(territories.get(i));
+      points[i] = getPointOnMap(mapData.getCenter(territories.get(i)), xOffset, yOffset, width, height);
     }
     if (routeDescription.getStart() != null) {
-      points[0] = routeDescription.getStart();
+      points[0] = getPointOnMap(routeDescription.getStart(), xOffset, yOffset, width, height);
     }
     if (routeDescription.getEnd() != null && numTerritories > 1) {
-      points[numTerritories - 1] = new Point(routeDescription.getEnd());
+      points[numTerritories - 1] =
+          getPointOnMap(routeDescription.getEnd(), xOffset, yOffset, width, height);
     }
     return points;
   }
 
   /**
+   * This method ensures that Points are drawn correctly, even on an infinite-scroll Map
```

Comment:
How does it ensure this? The comment is not very specific. This method would likely do well as its own object, ie, perhaps: `OnMapPointLocator`